apps/contributions/models.py

```
@@ -1383,6 +1384,28 @@ def update_subscription_amount(self, amount: int, donor_selected_amount: float)
         # Send amount updated email to contributor
         self.send_recurring_contribution_amount_updated_email()
 
+    def create_contributor_canceled_contribution_activity_log(self) -> ActivityLog | None:
+        """Create an activity log entry for a contributor-canceled contribution.
+
+        This is used to track the cancellation of a contribution by the contributor.
+        """
+        try:
+            if not (contributor := self.contributor):
+                logger.warning("Cannot create activity log for contribution %s because it has no contributor", self.pk)
+                return None
+            return ActivityLog.objects.create(
+                actor_content_object=contributor,
+                activity_object_content_object=self,
+                action=ActivityLog.CANCEL,
+            )
+        # We're generically catching on Exception here because we don't want callers to fail if something goes wrong
+        # creating an activity log.
+        except Exception:
```

Don't we need a #noqa here because we're catching `Exception` or is that not something our CI checks for? If not, should we enable that?

---

That's not something we look for. By default we do look for bare except via [Ruff E722](https://docs.astral.sh/ruff/rules/bare-except/).